### PR TITLE
fix(turbopack): Make node js use our sourcemap corectly

### PR DESCRIPTION
**Pull request by kdy1**
### Description

I need to make a repro for a question on node.js issue tracker.

### CODE REVIEW

**Comment by kdy1** [comment]
I filed https://github.com/orgs/nodejs/discussions/51675

File: crates/turbopack-core/src/source_map/mod.rs [hunk 1]
```diff
@@ -620,7 +600,20 @@ impl Serialize for CrateMapWrapper {
     fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
         use serde::ser::Error;
         let mut bytes = vec![];
-        self.0.to_writer(&mut bytes).map_err(Error::custom)?;
+        match &self.0 {
```
Comment (sokra) [inline crates/turbopack-core/src/source_map/mod.rs:603]: Why do we want to change the serialization logic?

File: crates/turbopack-core/src/source_map/mod.rs [hunk 2]
```diff
@@ -630,7 +623,15 @@ impl<'de> Deserialize<'de> for CrateMapWrapper {
         use serde::de::Error;
         let bytes = <&[u8]>::deserialize(deserializer)?;
         let map = DecodedMap::from_reader(bytes).map_err(Error::custom)?;
-        Ok(CrateMapWrapper(map))
+
```
Comment (sokra) [inline crates/turbopack-core/src/source_map/mod.rs:626]: Why do we want to change the deserialization logic?

File: crates/turbopack-core/src/source_map/mod.rs [hunk 3]
```diff
@@ -218,9 +218,9 @@ impl SourceMap {
                         )
                     })
                     .collect::<Vec<sourcemap::SourceMapSection>>();
-                Arc::new(CrateMapWrapper(DecodedMap::Index(SourceMapIndex::new(
-                    None, sections,
-                ))))
+                let map = SourceMapIndex::new(None, sections);
+                let map = map.flatten()?;
```
Comment (sokra) [inline crates/turbopack-core/src/source_map/mod.rs:222]: I don't think we should flatten the source map in general.

Browsers support indexed source maps so we can use them here.
Our own lookup supports indexed source maps.

Only node.js doesn't support it, so we should only change the emitted SourceMaps to be flattened.

File: crates/turbopack-core/src/source_map/mod.rs [hunk 4]
```diff
@@ -257,44 +257,24 @@ impl SourceMap {
...
+                        Some(map.0.clone()),
+                    ));
                 }
 
-                rope += "]
-}";
-
-                rope.build()
+                let map = SourceMapIndex::new(None, sections);
+                let map = CrateMapWrapper(DecodedMap::Regular(map.flatten()?));
```
Comment (sokra) [inline crates/turbopack-core/src/source_map/mod.rs:274]: See above

File: crates/turbopack-core/src/source_map/mod.rs (hunk 1 above)
Comment (kdy1) [reply crates/turbopack-core/src/source_map/mod.rs:603]: I did it while debugging, because I wasn't sure the code used for source map generation

**Comment by kdy1** [comment]
Closing as this PR is now redundant. The source map issue is fixed on the main branch, and we don't need to generate non-sectioned source map anymore.

**Pull request closed without merging**